### PR TITLE
chore: Switch CI to GitHub Actions

### PR DESCRIPTION
**Pull request by nschonni**
Travis is going to start limiting capacity for OSS projects https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
- Adds macOS testing
- Add Node 15 testing which has a failure as mentioned in https://github.com/igorshubovych/markdownlint-cli/pull/114#issuecomment-722506089

You can see the run https://github.com/nschonni/markdownlint-cli/actions/runs/348014552 since it looks like Actions isn't enabled on this repo yet

### CODE REVIEW

**Comment by nschonni** [comment]
OK, think I found an OK alternate for the failing test and the CI is happy again https://github.com/nschonni/markdownlint-cli/actions/runs/348047840

File: README.md [hunk 1]
```diff
@@ -111,11 +113,8 @@ A JS configuration file may internally `require` one or more npm packages as a w
 
 MIT © Igor Shubovych
 
-[travis-badge]: https://img.shields.io/travis/igorshubovych/markdownlint-cli/master.svg?label=linux
-[travis-url]: https://travis-ci.org/igorshubovych/markdownlint-cli
-
-[appveyor-badge]: https://img.shields.io/appveyor/ci/igorshubovych/markdownlint-cli/master.svg?label=windows
-[appveyor-url]: https://ci.appveyor.com/project/igorshubovych/markdownlint-cli
+[actions-badge]: https://github.com/igorshubovych/markdownlint-cli/workflows/Test/badge.svg
```
Comment (DavidAnson) [inline README.md:116]: Please use the versions that lock to "master" branch.

File: test/test.js [hunk 2]
```diff
@@ -86,7 +86,7 @@ test('linting of incorrect Markdown via npm run file fails with eol', async t =>
     await execa('npm', ['run', 'invalid'], {stripFinalNewline: false});
     t.fail();
   } catch (error) {
-    t.regex(error.stderr, /\nnpm ERR! code ELIFECYCLE/);
+    t.regex(error.stderr, /MD\d{3}.*\nnpm ERR!/);
```
Comment (DavidAnson) [inline test/test.js:89]: Thanks!

**Review by DavidAnson — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+name: Test
```
[the remark this answers is not in the document]
Comment (nschonni) [reply .github/workflows/ci.yml]: Done

File: README.md (hunk 1 above)
Comment (nschonni) [reply README.md]: Looks like it uses "Default branch" if you don't specify it, but I pinned it to master anyway

**Review by DavidAnson — APPROVED** [review]
Great!

**Comment by nschonni** [comment]
Hooray, looks like it's working https://github.com/igorshubovych/markdownlint-cli/actions/runs/348495120

**Pull request merged**